test/fluree/db/query/misc_queries_test.clj

```diff
@@ -239,60 +237,63 @@
-                  [:ex/User :id "http://example.org/ns/User"]
-                  [:rdfs/Class :id "http://www.w3.org/2000/01/rdf-schema#Class"]
-                  [:rdf/type :id "http://www.w3.org/1999/02/22-rdf-syntax-ns#type"]
-                  [:f/t :id "https://ns.flur.ee/ledger#t"]
-                  [:f/size :id "https://ns.flur.ee/ledger#size"]
-                  [:f/flakes :id "https://ns.flur.ee/ledger#flakes"]
-                  [:f/defaultContext :id "https://ns.flur.ee/ledger#defaultContext"]
-                  [:f/branch :id "https://ns.flur.ee/ledger#branch"]
-                  [:f/alias :id "https://ns.flur.ee/ledger#alias"]
-                  [:f/data :id "https://ns.flur.ee/ledger#data"]
-                  [:f/address :id "https://ns.flur.ee/ledger#address"]
-                  [:f/v :id "https://ns.flur.ee/ledger#v"]
-                  ["https://www.w3.org/2018/credentials#issuer" :id "https://www.w3.org/2018/credentials#issuer"]
-                  [:f/time :id "https://ns.flur.ee/ledger#time"]
-                  [:f/message :id "https://ns.flur.ee/ledger#message"]
-                  [:f/previous :id "https://ns.flur.ee/ledger#previous"]
-                  [:id :id "@id"]
-                  ["fluree:commit:sha256:bgc3scabiyx6v22isv7kchxpktilynx2xgcx3mtau6dxoeph5dwg" :id "fluree:commit:sha256:bgc3scabiyx6v22isv7kchxpktilynx2xgcx3mtau6dxoeph5dwg"]
-                  ["fluree:commit:sha256:bgc3scabiyx6v22isv7kchxpktilynx2xgcx3mtau6dxoeph5dwg" :f/time 720000]
-                  ["fluree:commit:sha256:bgc3scabiyx6v22isv7kchxpktilynx2xgcx3mtau6dxoeph5dwg" "https://www.w3.org/2018/credentials#issuer" "did:fluree:TfCzWTrXqF16hvKGjcYiLxRoYJ1B8a6UMH6"]
-                  ["fluree:commit:sha256:bgc3scabiyx6v22isv7kchxpktilynx2xgcx3mtau6dxoeph5dwg" :f/v 0]
-                  ["fluree:commit:sha256:bgc3scabiyx6v22isv7kchxpktilynx2xgcx3mtau6dxoeph5dwg" :f/address "fluree:memory://87d951b0c27423f8e07de484e2e68bcb36db5baca9ea3aca05c86a1a19faf4b5"]
-                  ["fluree:commit:sha256:bgc3scabiyx6v22isv7kchxpktilynx2xgcx3mtau6dxoeph5dwg" :f/data "fluree:db:sha256:btjaah6rs4wff6m2bd7n2rps6rkjpvnktbf3lxvwsijvbcdv3asw"]
-                  ["fluree:commit:sha256:bgc3scabiyx6v22isv7kchxpktilynx2xgcx3mtau6dxoeph5dwg" :f/alias "query/everything"]
-                  ["fluree:commit:sha256:bgc3scabiyx6v22isv7kchxpktilynx2xgcx3mtau6dxoeph5dwg" :f/branch "main"]
-                  ["fluree:commit:sha256:bgc3scabiyx6v22isv7kchxpktilynx2xgcx3mtau6dxoeph5dwg" :f/defaultContext "fluree:context:b6dcf8968183239ecc7a664025f247de5b7859ac18cdeaace89aafc421eeddee"]]
-                 @(fluree/query db* {:select ['?s '?p '?o]
-                                     :where  [['?s '?p '?o]]}))))))))
+        (let [db*    @(fluree/commit! ledger db)
+              result @(fluree/query db* {:select ['?s '?p '?o]
+                                         :where  [['?s '?p '?o]]})]
+          (is (pred-match?
```

Comment:
If the no-predicates-in-a-pred-match test in the history-test namespace above was accidental, I think this test would be a good candidate for the did-the-hashes-change test.

Reply:
Actually, I think I'd like to consider this a bit more later on. There are reasons why the hashes in this test could still change as we stabilize things, and updating even just these is a poor use of our time.

On the other hand, it could make sense to add more specific unit tests that check that our hashes are stable for the same data (this seems too high level for that; more integration test-y) and then–once our in-storage data formats are stabilized–we can add in some specific integration / regression testing that they don't change (rather than it being a side effect of a test for `?s ?p ?o` queries).

Those are my thoughts on the matter, at least. :)

Reply:
I think we need at least one stable-hashes test so we don't accidentally introduce some non-deterministic input to the hashes. I don't have a strong preference over whether it's this one or a more targeted one, as long as we balance the expedience of making internal changes with the ability to catch accidental non-determinism.

I've been saved by the explicit hashes at least once when it turned out I was hashing the `pr-str` of a map, and the order of the keys wasn't deterministic. If we don't have any stable-hash tests, it will be harder to figure out what the source of the non-determinism is down the road.

Reply:
What do you think of the approach in [0adead6](https://github.com/fluree/db/pull/506/commits/0adead6a8a2e68963a4e2a38cf0af7f7abf268b6)? Any other hashes we should test in there?

Reply:
Looks like you got all of the important ones to me!